weave-js/src/components/Button/Button.tsx

```diff
@@ -4,153 +4,153 @@
  * https://www.figma.com/file/01KWBdMZg5QM9SRS1pQq0z/Design-System----Robot-Styles?type=design&node-id=5956-31813&mode=design&t=Gm4WWGWwgjdfUUTe-0
  */
 
-import {TooltipContentProps} from '@radix-ui/react-tooltip';
+import {type TooltipContentProps} from '@radix-ui/react-tooltip';
 import classNames from 'classnames';
-import React, {ReactElement, useState} from 'react';
+import React, {
+  CSSProperties,
+  forwardRef,
+  type ReactElement,
+  useState,
+} from 'react';
 import {twMerge} from 'tailwind-merge';
 
-import {Icon, IconName} from '../Icon';
+import {Icon, type IconName} from '../Icon';
 import * as Tooltip from '../RadixTooltip';
 import {Tailwind} from '../Tailwind';
-import {ButtonSize, ButtonVariant} from './types';
+import type {ButtonSize, ButtonVariant} from './types';
 
-export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
-  /**
-   * Shorthand alias for `startIcon` since most button icons are startIcons.
-   * If both props are provided, `startIcon` value will be used out of the two.
-   */
+type IconButtonProps = {
   icon?: IconName;
-  /**
-   * Icon displayed before the children
-   */
+  children?: never;
+  startIcon?: never;
+  endIcon?: never;
+  tooltip: string;
+};
+
+type LabelButtonProps = {
+  icon?: never;
+  children: ReactElement | string;
   startIcon?: IconName;
-  /**
-   * Icon displayed after the children
-   */
   endIcon?: IconName;
+  tooltip?: string;
+};
+
+export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
   size?: ButtonSize;
   variant?: ButtonVariant;
   children?: ReactElement | string;
   active?: boolean;
   tooltip?: string;
   tooltipProps?: TooltipContentProps;
   twWrapperStyles?: React.CSSProperties;
+} & (IconButtonProps | LabelButtonProps);
+
+const sizeMap = {
+  small:
+    'gap-6 px-6 py-3 text-sm leading-[18px] [&_svg]:h-16 [&_svg]:w-16 h-24 w-24 p-0',
+  medium: 'gap-10 px-10 py-4 text-base [&_svg]:h-18 [&_svg]:w-18 h-32 w-32 p-0',
+  large: 'gap-8 px-12 py-8 text-base h-40 w-40 p-0',
 };
 
-export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+const variantMap = {
+  primary: 'bg-teal-500 text-white hover:bg-teal-450',
+  secondary:
+    'bg-oblivion/[0.05] dark:bg-moonbeam/[0.05] text-moon-800 dark:text-moon-200 hover:bg-teal-300/[0.48] hover:text-teal-600 dark:hover:bg-teal-700/[0.48] dark:hover:text-teal-400',
+  ghost:
+    'bg-oblivion/[0.05] dark:bg-moonbeam/[0.05] text-moon-800 dark:text-moon-200 hover:bg-teal-300/[0.48] hover:text-teal-600 dark:hover:bg-teal-700/[0.48] dark:hover:text-teal-400',
+  quiet:
+    'text-moon-500 hover:text-moon-800 dark:hover:text-moon-200 hover:bg-oblivion/[0.05] dark:hover:bg-moonbeam/[0.05]',
+  destructive: 'bg-red-500 text-white hover:bg-red-450',
+};
+
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   (
     {
       size = 'medium',
       variant = 'primary',
       icon,
-      startIcon = icon,
+      startIcon,
```

Comment:
This may be convenient for consumers but makes for inconsistency between call sites leading to code that is harder to refactor and reason with.

Reply:
⚠️ This is a breaking change.

It would be nice if weave was a package so we could optionally upgrade in a PR that updates the app with the necessary call site changes. I'm open to suggestions on how to make breaking changes without actually breaking the app.

Reply:
you wouldn't need to block deploys, this PR would just be blocked until all the call sites are updated